Library/ViewModels/BackerDashboardViewModel.swift

```diff
@@ -0,0 +1,239 @@
+
+  /// Emits when to navigate to Settings.
+  var goToSettings: Signal<(), NoError> { get }
+
+  /// Emits an index to pin the indicator view to a particular button view with or without animation.
+  var pinSelectedIndicatorToPage: Signal<(Int, Bool), NoError> { get }
+
+  /// Emits an attributed string for the saved button title label.
+  var savedButtonTitleText: Signal<String, NoError> { get }
+
+  /// Emits a boolean whether saved projects container should be hidden or not.
+  var savedProjectsAreHidden: Signal<Bool, NoError> { get }
+
+  /// Emits an index of the selected button to update all button selected states.
+  var setSelectedButton: Signal<Int, NoError> { get }
+
+  /// Emits when should scroll to the project item or row position.
+  var scrollToProject: Signal<Int, NoError> { get }
+
+  /// Emits a boolean whether the sort bar is hidden or not.
+  var sortBarIsHidden: Signal<Bool, NoError> { get }
+}
+
+public protocol BackerDashboardViewModelType {
+  var inputs: BackerDashboardViewModelInputs { get }
+  var outputs: BackerDashboardViewModelOutputs { get }
+}
+
+public final class BackerDashboardViewModel: BackerDashboardViewModelType, BackerDashboardViewModelInputs,
+  BackerDashboardViewModelOutputs {
+
+  // swiftlint:disable:next function_body_length
+  public init() {
+    let user = Signal.merge(
+      self.viewDidLoadProperty.signal,
+      self.viewWillAppearProperty.signal.filter(isFalse).skip(first: 1).ignoreValues()
+      )
+      .switchMap { _ in
+        AppEnvironment.current.apiService.fetchUserSelf()
+          .prefix(SignalProducer([AppEnvironment.current.currentUser].compact()))
+          .demoteErrors()
+    }
+
+    self.avatarURL = user.map { URL(string: $0.avatar.large ?? $0.avatar.medium) }
+
+    self.backedButtonTitleText = user
+      .map { user in
+        localizedString(
+          key: "projects_count_backed",
+          defaultValue: "%{projects_count} backed",
+          count: user.stats.backedProjectsCount ?? 0,
+          substitutions: ["projects_count": Format.wholeNumber(user.stats.backedProjectsCount ?? 0)]
+        )
+        .replacingOccurrences(of: " ", with: "\n")
+    }
+
+    self.backerLocationText = user.map { $0.location?.displayableName ?? "" }
+
+    self.backerNameText = user.map { $0.name }
+
+    self.configureBackedProjectsController = self.viewDidLoadProperty.signal
+      .map { .backed }
+
+    self.configureSavedProjectsController = self.viewDidLoadProperty.signal
+      .map { .saved }
+
+    self.savedButtonTitleText = user.map { user in
+      localizedString(
+        key: "projects_count_backed",
+        defaultValue: "%{projects_count} saved",
+        count: user.stats.starredProjectsCount ?? 0,
+        substitutions: ["projects_count": Format.wholeNumber(user.stats.starredProjectsCount ?? 0)]
+      )
+      .replacingOccurrences(of: " ", with: "\n")
+    }
+
+    self.backedProjectsAreHidden = Signal.merge(
```

Comment:
tho all these outputs are easy to read, it will be nice to just let the page view controller take care of all of it for us